packages/flutter/test/rendering/editable_test.dart

```diff
@@ -587,4 +592,64 @@ void main() {
     editable.layout(BoxConstraints.loose(const Size(1000.0, 1000.0)));
     expect(editable.maxScrollExtent, equals(10));
   });
+
+  test('attach and detach correctly handle gesture', () {
+    _GestureBindingSpy();
+    final TextSelectionDelegate delegate = FakeEditableTextState();
+    final RenderEditable editable = RenderEditable(
+      backgroundCursorColor: Colors.grey,
+      selectionColor: Colors.black,
+      textDirection: TextDirection.ltr,
+      cursorColor: Colors.red,
+      offset: ViewportOffset.zero(),
+      textSelectionDelegate: delegate,
+      text: const TextSpan(
+        text: 'test',
+        style: TextStyle(
+          height: 1.0, fontSize: 10.0, fontFamily: 'Ahem',
+        ),
+      ),
+      startHandleLayerLink: LayerLink(),
+      endHandleLayerLink: LayerLink(),
+      selection: const TextSelection(
+        baseOffset: 0,
+        extentOffset: 3,
+        affinity: TextAffinity.upstream,
+      ),
+      onSelectionChanged: (_, __, ___) { },
+    );
+    final PipelineOwner owner = PipelineOwner(onNeedVisualUpdate: () { });
+    editable.attach(owner);
+    // This should register pointer into GestureBinding.instance.pointerRouter.
+    editable.handleEvent(const PointerDownEvent(), BoxHitTestEntry(editable, const Offset(10,10)));
+    // ignore: avoid_as
```

Comment:
We generally avoid using as cast, you can re cast it first before doing the count
```dart
final _PointerRouterSpy spy = GestureBinding.instance.pointerRouter
expect(spy. routeCount, greaterThan(0));
editable.detach();
expect(spy. routeCount, 0);
```